src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -664,6 +666,13 @@ def _prep_net_for_train(self) -> None:
         self.resume_from_remote_sg_logger = core_utils.get_param(self.training_params, "resume_from_remote_sg_logger", False)
         self.load_checkpoint = self.load_checkpoint or self.external_checkpoint_path is not None or self.resume_from_remote_sg_logger
 
+        if self.training_params.torch_compile:
+            logger.info("Using torch.compile feature. Compiling model. This may take a few minutes")
+            self.net = torch.compile(self.net, mode=self.training_params.torch_compile_mode)
```

Comment:
✅ Added check for torch 2.0+
➖ Adding try/catch is not quite possible because compile operates on the first forward & backward as far as I can tell and more importantly for some reason it does the second compile during validation 🤷 I'm not sure whether this is an expected behavior or not, but this is what I'm observing. Long story short, we can see whether compile actually worked or not only inside the train loop. Do we really want to add try/catch everywhere? I feel this would complicate the logic quite a lot
✅ Added `torch_compile_loss`, but tried on PPYoloE loss with no luck

Reply:
Alrighty, thanks